Wikipedia/Code/WMFCompassView.m

```diff
@@ -223,4 +223,21 @@ - (void)drawCompassLinesInContext:(CGContextRef)ctx
     }
 }
 
+#pragma mark - Accessibility
+
+- (BOOL)isAccessibilityElement {
+    return YES;
+}
+
+- (NSString *)accessibilityLabel {
+    double positive_angle = self.angleRadians >= 0 ? self.angleRadians : self.angleRadians + 2*3.14;
+    NSInteger clock_direction = lroundf(positive_angle / (3.14 / 6));
+    if (clock_direction == 0) {
+        clock_direction = 12;
+    }
+    NSString* label = MWLocalizedString(@"compass-direction", nil);
+    label = [label stringByReplacingOccurrencesOfString:@"$1" withString:[NSString localizedStringWithFormat:@"%@", @(clock_direction)]];
```

Comment:
i think the `localizedStringWithFormat` is unnecessary here, since all the values are "**N** o'clock" and don't vary based on units of measurement or 12/24-hour time.